cpp/src/gandiva/precompiled/timestamp_arithmetic.cc

```diff
@@ -17,6 +17,7 @@
 
 #include "./epoch_time_point.h"
 
+static int days_in_a_month[] = {31, 28, 31, 30, 31, 30, 31, 31, 30, 31, 30, 31};
```

Comment:
one option might be to have a 2 by 12 array where the first row is days in months in non-leap years, and the second is days in months with leap years. you can probably then avoid some of the repeated branching code below.

Reply:
@cpcloud I applied your suggestion in that commit https://github.com/apache/arrow/pull/11259/commits/56b5ed615754bb222410f578af8c7c4e72a2ed7f check if it is good